### ISSUE TITLE
Clicking on some pinned tabs swaps the workspace

### ISSUE THREAD

**Issue by neurokitti** [zen-browser/desktop#2086]
### Captchas

- [X] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

https://github.com/user-attachments/assets/6d370260-24e5-4a40-b052-cef81703c4cf

when i click on some pinned tabs it swaps me to a different workspace. I am not sure why.

### Reproducible?

- [ ] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-8a (twilight)

### What platform are you seeing the problem on?

Windows

**Comment by ilahazs** [comment]
still exist

**Comment by ilahazs** [comment]
what do you mean by completed?

version : 1.0.1-a.9 (64-bit)

https://github.com/user-attachments/assets/d4eb5661-07dd-4d8b-8372-81fae7fe0c1d

**Comment by mmattq** [comment]
Happens to me as well, but only after upgrade to version 1.0.1-a.9.
OS: Arch Linux

**Comment by kristijanribaric** [comment]
Did anyone figure out the steps to reproduce this?

**Comment by ilahazs** [comment]
Clicking on pinned tab changing the workspace randomly, not always like click on 1st pinned tab changing to 1st workspace but the behavior is a bit random

**Comment by mmattq** [comment]
> Clicking on pinned tab changing the workspace randomly, not always like click on 1st pinned tab changing to 1st workspace but the behavior is a bit random

I don't think it's random, it switches to the workspace where the tab was originally pinned.

**Comment by mr-cheffy** [comment]
Does this still happen in version a10?

**Comment by ilahazs** [comment]
> > Clicking on pinned tab changing the workspace randomly, not always like click on 1st pinned tab changing to 1st workspace but the behavior is a bit random
> 
> I don't think it's random, it switches to the workspace where the tab was originally pinned.

yeah you're right

**Comment by mmattq** [comment]
> Does this still happen in version a10?

It's fixed for me, thanks!

**Comment by mmattq** [comment]
I found a new bug related to this one. When I pin a tab in a workspace (e.g., WORK), switch to another workspace (e.g., PERSONAL) and then delete the WORK workspace, the pinned tab from WORK workspace is also deleted. But maybe this should be a separate issue?

**Comment by kristijanribaric** [comment]
I don't see how that is a bug, if you delete a workspace of course all tabs in that workspace will be removed.

**Comment by mmattq** [comment]
> I don't see how that is a bug, if you delete a workspace of course all tabs in that workspace will be removed.

I thought the pinned tab would stay open, but if that's not a bug, then it's fine.

**Comment by ilahazs** [comment]
> Does this still happen in version a10?

it's fixed in a 10, tysm.

**Comment by mr-cheffy** [comment]
I mean, what else should it do?

**Comment by neurokitti** [comment]
yup! fixed for me! thanks ❤️
